src/bots/HighSellVolumeBot.js

```diff
@@ -0,0 +1,255 @@
+  process.env.BUY_LIQUIDITY_BOT_CHECK_TIME_MS || (60 * 1000) // 1 min
+
+class HighSellVolumeBot extends Bot {
+  constructor ({
+    name,
+    // eventBus,
+    // liquidityService,
+    dxInfoService,
+    marketService,
+    botAddress,
+    markets,
+    slackClient,
+    botTransactionsSlackChannel,
+    // buyLiquidityRules,
+    notifications,
+    checkTimeInMilliseconds = ENSURE_LIQUIDITY_PERIODIC_CHECK_MILLISECONDS
+  }) {
+    super(name)
+
+    // this._eventBus = eventBus
+    // this._liquidityService = liquidityService
+    this._dxInfoService = dxInfoService
+    this._marketService = marketService
+    this._botAddress = botAddress
+    this._markets = markets
+    this._slackClient = slackClient
+    this._botTransactionsSlackChannel = botTransactionsSlackChannel
+    // this._buyLiquidityRules = buyLiquidityRules
+    this._notifications = notifications
+    this._checkTimeInMilliseconds = checkTimeInMilliseconds
+
+    this._lastCheck = null
+    this._lastBuy = null
+    this._lastError = null
+  }
+
+  async _doStart () {
+    logger.debug({ msg: 'Initialized bot: ' + this.name })
+
+    this._markets.forEach(market => {
+      const sellToken = market.tokenA
+      const buyToken = market.tokenB
+      this._doRoutineLiquidityCheck(sellToken, buyToken)
+    })
+
+    // Check the liquidity periodically
+    setInterval(() => {
+      this._markets.forEach(market => {
+        const sellToken = market.tokenA
+        const buyToken = market.tokenB
+        this._doRoutineLiquidityCheck(sellToken, buyToken)
+      })
+    }, this._checkTimeInMilliseconds)
+  }
+
+  async _doStop () {
+    logger.debug({ msg: 'Bot stopped: ' + this.name })
+  }
+
+  async _doRoutineLiquidityCheck (sellToken, buyToken) {
+    // Do ensure liquidity on the market
+    auctionLogger.debug({
+      sellToken,
+      buyToken,
+      msg: "Doing a routine check. Let's see if we will be able to ensure the liquidity"
+    })
+    return this._checkBuyLiquidity({
+      sellToken,
+      buyToken,
+      from: this._botAddress
+    })
+  }
+
+  async _checkBuyLiquidity ({ sellToken, buyToken, from }) {
+    this._lastCheck = new Date()
+    let liquidityWasChecked
+    try {
+      // We must check both auction sides
+      await this._checkEnoughBalance({ sellToken, buyToken, from })
+      await this._checkEnoughBalance({ sellToken: buyToken, buyToken: sellToken, from })
+
+      liquidityWasChecked = true
+    } catch (error) {
+      this.lastError = new Date()
+      liquidityWasChecked = false
+      this._handleError(sellToken, buyToken, error)
+    }
+
+    return liquidityWasChecked
+  }
+
+  async _checkEnoughBalance ({ sellToken, buyToken, from }) {
+    logger.debug('Checking enough balance for %s-%s in %s', sellToken, buyToken, from)
+    const auctionSellVolumePromise = this._dxInfoService.getSellVolume({
+      sellToken, buyToken })
+    const accountBalancesPromise = this._dxInfoService.getBalances({
```

Comment:
We don't need to check all the balances, we can check for the current token pair

Reply:
Yes, we can save some time there and simplify the code

Good point